rbd/rbd.go

```diff
@@ -808,6 +808,35 @@ func (image *Image) WriteAt(data []byte, off int64) (n int, err error) {
 	return ret, err
 }
 
+// WriteSame repeats writing data from startpoint off until n bytes have been written.
+//
+// Implements:
+//  ssize_t rbd_writesame(rbd_image_t image, uint64_t ofs, size_t len,
+//                        const char *buf, size_t data_len, int op_flags);
+func (image *Image) WriteSame(ofs, n uint64, data []byte) (int64, error) {
+	var err error
+
+	if err = image.validate(imageIsOpen); err != nil {
+		return 0, err
+	}
+
+	if len(data) == 0 {
+		return 0, nil
+	}
+
+	ret := C.rbd_writesame(image.image,
+		C.uint64_t(ofs),
+		C.uint64_t(n),
+		(*C.char)(unsafe.Pointer(&data[0])),
+		C.size_t(len(data)),
+		0 /* LIBRADOS_OP_FLAG */)
```

Comment:
Without looking up the source to librbd I don't know if this parameter is used or unused in ceph, or what it does in the librbd API, etc. If it's not necessary and we're not going to expose it, it may be better to expand on the comment to explain why.

Reply:
These flags are [listed in librados.h](https://github.com/ceph/ceph/blob/master/src/include/rados/librados.h#L67). I am in doubt if adding them is helpful here, or not. It should probably be possible to pass them.

Reply:
OK, I've encountered those flags when I was working on #315. All I can figure is that there's a use case for possibly passing the _EXCL flag or the FALLOC flags in some cases for this call and that this is implemented via the write op functions. (Otherwise why would this argument be here?)
That all said, I'm fine with not wrapping it as long as we comment why we're not going to. :-)

Reply:
After some more consideration, I think my preference is to be able to pass the flags like the librbd API provides.